File: SWEA/D2/2001. 파리 퇴치/파리 퇴치.cpp
#include<iostream>
#include <vector>
using namespace std;

int calc(vector<vector<int>> &v, int x, int y, int m)
{
    int ret = 0;
    
    for(int i = x; i < (x+m); i++)
    {
        for(int j = y; j < (y + m); j++)
        {
            ret += v[i][j];
        }
    }
    
    return ret;
}

int main(int argc, char** argv)
{
	int test_case;
	int T;
	cin>>T;

	for(test_case = 1; test_case <= T; ++test_case)
	{
        int n, m;
        cin >> n >> m;
        
        vector<vector<int>> v(n);
        
        for(int i = 0; i < n; i++)
        {
            for(int j = 0; j < n; j++)
            {
                int input;
                cin >> input;
                v[i].push_back(input);
            }
        }
        
        int maxnum = 0;
        for(int i = 0; i <= (n- m); i++)
        {
            for(int j = 0; j <= (n - m); j++)
            {
                maxnum = max(maxnum, calc(v, i, j, m));
            }
        }
        
        cout << "#" << test_case << " ";
        cout << maxnum << '\n';
	}
	return 0;//정상종료시 반드시 0을 리턴해야합니다.
}

File: SWEA/D2/2001. 파리 퇴치/파리 퇴치.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Solution {
    static int result;
    static void calc(int[][] map, int N, int M, int x, int y) {
        int cnt = 0;
        for (int i = x; i < x + M; i++) {
            for (int j = y; j < y + M; j++) {
                if(i >= 0 && i < N && j >= 0 && j < N)
                    cnt += map[i][j];
            }
        }
        result = Math.max(result, cnt);
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int T = Integer.parseInt(br.readLine());

        for (int test_case = 1; test_case <= T; test_case++) {
            String[] NM = br.readLine().split(" ");
            int N = Integer.parseInt(NM[0]);
            int M = Integer.parseInt(NM[1]);

            result = 0;

            int[][] map = new int[N][N];
            for (int i = 0; i < N; i++) {
                String[] input = br.readLine().split(" ");
                for (int j = 0; j < N; j++) {
                    int num = Integer.parseInt(input[j]);
                    map[i][j] = num;
                }
            }

            for (int i = 0; i < N; i++) {
                for (int j = 0; j < N; j++) {
                    calc(map, N, M, i, j);
                }
            }

            sb.append("#").append(test_case).append(" ").append(result).append("\n");
        }
        System.out.println(sb.toString());
    }
}
